### ISSUE TITLE
faulty duration time in SCID

### ISSUE THREAD

**Issue by tissatussa** [algerbrex/blunder#12]
Recently i was testing the following position on several engines. It's from the game Giri vs. Firouzja at the Tata Steel Masters tournament 2021, eg. see https://chess24.com/en/watch/live-tournaments/tata-steel-masters-2021/12/1/4

After 60...Nf4 this position was on the board :

![blunder_v7 6 0_strange_time](https://user-images.githubusercontent.com/1109281/148643699-19040c28-ed2a-4884-89f6-24ef8da16ef8.png)

r7/P5R1/2k5/8/1P3n1P/K7/6P1/8 w - - 0 18

White to play. The only winning move is Ka4. Not many engines find this. Using multipv 1, Blunder finds this move as best after 2 minutes (i'm using Xubuntu 20.04 on a modern notebook). This position and its "solution" are interesting, but i open this Issue because Blunder seems to output a faulty time, see the yellow-red marker dot i put in the image - i used SCID v4.7.1 for this.

Normally these times in SCID are increasing, they represent the total time. But it looks like Blunder outputs how much time the evaluation took for each depth, not the total time !?

**Comment by algerbrex** [comment]
Hi @tissatussa. Thanks for bringing up this issue. Looks like you're correct and I've misunderstood the UCI protocol, like every other engine I test displays the total time, not the time for each depth like I do.

The fix is pretty simple actually, so I'll go ahead and merge it into the dev branch and this issue will be closed when the next version of Blunder is released hopefully two-ish weeks from now, depending on the progress I can make.

And interesting position. Not sure why it's taking Blunder so long. Likely pruning or reductions happening along the path to the correct line.

**Comment by tissatussa** [comment]
Yes, i liked to mention this Issue .. i can imagine it's not hard to solve it ..

About this position : most strong engines find Ka4 rather quickly, but not all .. eg. Komodo 13 MCTS finds it only after 62 sec. And Toga II and III both do not find it, although they're about 2800 and 3100 rating. Could you trace why Blunder does not find it quickly ? I wonder .. i have no idea how to do that, i'm not into chess engine programming, i'm just a user of engines.

**Comment by algerbrex** [comment]
Cool. And I could probably trace why Blunder doesn't choose Ka4 initially, but it'd probably be a little tedious. I'd have to basically track the line caused by Ka4 and see what leaf evaluation the engine ultimately returned in the qsearch. 

I'm ok with living with Blunder finding the move in a minute or two now though. That's the price of pruning and reductions sometimes :-)

**Comment by tissatussa** [comment]
wow !  i'm pleased you elaborate on this .. i guess you're a fanatic chess player yourself, knowing many aspects of our game .. especially i like `That's the price of pruning and reductions sometimes` because i think pruning (as i understand & imagine it) is at the core of (most) engines : not spend calculation time in "irrelevant" nodes, am i right?

do you know the existence of `p4wn chess` ? it's tiny javascript below 5 Kb (including board graphics!) that plays very fast and rather good .. amazing .. the creator must have put his chess knowledge into some nifty "rules of thumb" which suit many positions : http://p4wn.sourceforge.net

i encountered that engines can better find a "puzzle best move" when using multipv, although each depth will take more time. Without multipv the engine will never find the move. Setting multipv high (eg. 20 or even 30) may be needed for the engine to EVER find the puzzle move - in that case, i wonder if this is a realistic scenario ..

also the UCI option "UCI_AnalyseMode" exists, type check .. according to the README : _the engine wants to behave differently when analysing or playing a game. For example when playing it can use some kind of learning. This is set to false if the engine is playing a game, otherwise it is true._  this option maybe set automatically .. that's not clear to me .. and not every engine has this option .. and what does it do ? Can we see a difference, eg. by using test positions ?

i could write more about this, but a new Issue may be better ..

was it coincidence your dev version can solve my "puzzle" ?
anyhow, congrats with your new dev !

**Comment by algerbrex** [comment]
Thanks! I'm an okay-ish chess player but getting better. And yes, pruning and reductions are very ubiquitous in modern chess programs, and are actually quite human, in the sense that they're trying to help the computer avoid thinking about lines of play that are likely stupid or pointless. I have no idea how Carlsen or Firouzja see the board, but I do know it's likely they don't spend ten minutes analyzing a pointless queen sac!

And I haven't heard of p4wn before. It looked like an interesting project, and I was able to beat it on its default difficulty. Maybe I'll try to test how strong it is on its highest difficulty level. 

And multi-pv is something I'm not very knowledgeable on yet myself, but conceptually, it's just the engine giving what it thinks are the n-best lines of play for both sides. So what's likely happening for many engines is that the line starting with Ka4 isn't regarded as a _bad_ line of play, just not one as good as others. So it'll show up as maybe the second or third best option when multi-pv is enabled, when in reality of course, it's the only winning line.

And for UCI_AnalyseMode, I think what exactly setting that to true entails is different for every engine. For example, whenever I get around to implementing the feature, I'll likely try to make my pruning more conservative so it can give a more accurate analysis of a position.

Thanks again a lot for the very useful bug report! Please don't hesitate to reach out again with another issue if you see anything else fishy in Blunder!
